### PR TITLE
feat: /learn page redesign with tabbed layout, model selector, and post-eval suggestions

### PR DESCRIPTION
**Pull request by William-Hill**
## Summary

- **Restructure /learn into 3 tabs** — Compare (default), Learn, Build — so returning users get interactive tools immediately instead of scrolling past educational content
- **Add hot-swappable model selector** — Pipeline A vs Pipeline B dropdowns for comparing any model configurations, with server-side validation against available models
- **Add rules-based post-eval suggestions** — after eval harness runs, maps failing metrics to actionable training data recommendations (e.g., `entity_f1 < 0.80` → "Add training pairs with more diverse entity types")
- **Add `POST /api/eval-runs/{id}/analyze` endpoint** — on-demand suggestion generation, idempotent with `?force=true` override
- **Add `--analyze` and `--analyze-existing` CLI flags** to eval harness for suggestion generation from the terminal

LLM-powered analysis (spec Section 5.2) is deferred to a follow-up PR — the `suggestions.llm_analysis` field is wired through end-to-end (DB → API → UI) so the follow-up only needs to implement the Claude API call itself.

## Test plan

- [ ] 579 Python tests passing (26 new tests across suggestions, schemas, endpoints, CLI flags)
- [ ] Next.js build clean, ESLint clean
- [ ] Visit `/learn` — Compare tab loads by default with model selector and eval metrics
- [ ] Switch to Learn tab — 7 educational sections render correctly
- [ ] Switch to Build tab — tutorials and slide deck link display
- [ ] Deep link `#learn` / `#build` / `#compare` — correct tab activates
- [ ] Run `python -m scripts.training.run_eval_harness --persist` — suggestions print after metrics
- [ ] Run `python -m scripts.training.run_eval_harness --analyze-existing latest` — loads and updates existing run
- [ ] Apply migration: `ALTER TABLE model_eval_runs ADD COLUMN IF NOT EXISTS suggestions JSONB`

## Summary by CodeRabbit

* **New Features**
  * Post-evaluation suggestion engine persisted with timestamps and shown after runs; API/UI can (re)analyze and refresh suggestions.
  * Eval runs now expose run IDs and suggestion payloads.
  * Explicit Pipeline A vs Pipeline B model selectors; model metadata includes type/version.
  * Learn page converted to tabs with new Build tab and tutorials; suggestions panel added with LLM analysis toggle and analyze action.

* **Chores**
  * Database migration to add suggestions column.

* **Tests**
  * New tests covering analysis endpoint, suggestion generation, model metadata, compare behavior, and CLI analyze flags.

### CODE REVIEW

File: tests/test_app_schemas.py [hunk 1]
```diff
@@ -41,3 +41,66 @@ def test_empty_question_rejected(self):
...
         with pytest.raises(ValidationError):
             QueryRequest(question="   ")
+
+
+from d4bl.app.schemas import (
+    CompareRequest,
+    EvalRunItem,
+    SuggestionItem,
+    AnalyzeResponse,
+)
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply tests/test_app_schemas.py]: Fixed in 1212e49b.

File: tests/test_compare_endpoint.py [hunk 2]
```diff
@@ -153,3 +153,85 @@ async def test_compare_same_model_returns_error(self):
...
+                    "prompt": "test",
+                    "pipeline_a_parser": "nonexistent-model",
+                    "pipeline_a_explainer": "mistral",
+                    "pipeline_b_parser": "mistral",
+                    "pipeline_b_explainer": "mistral",
+                })
+
+        assert resp.status_code == 400
+        assert "nonexistent-model" in resp.json()["detail"]
+        test_app.dependency_overrides.clear()
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply tests/test_compare_endpoint.py]: Fixed in 1212e49b.

File: ui-nextjs/components/learn/EvalMetricsPanel.tsx [hunk 3]
```diff
@@ -97,6 +98,14 @@ function TaskCard({ runs }: { runs: EvalRunItem[] }) {
...
             </div>
           )}
+          <SuggestionsPanel
+            suggestions={r.suggestions as Suggestions | null}
+            runId={r.id ?? ''}
+            onAnalyze={async (id) => {
+              await analyzeFailures(id);
+              await onRefresh();
+            }}
+          />
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply ui-nextjs/components/learn/EvalMetricsPanel.tsx]: Fixed in 1212e49b.

File: ui-nextjs/components/learn/LearnTabs.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    const applyHash = (h: string) => {
+      if (h && tabIds.includes(h)) {
+        setActiveTab(h);
+        setMounted((prev) => (prev.has(h) ? prev : new Set([...prev, h])));
+      }
+    };
+
+    // Read hash on mount (client-only) via microtask to satisfy lint rule
+    const initialHash = window.location.hash.replace("#", "");
+    queueMicrotask(() => applyHash(initialHash));
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply ui-nextjs/components/learn/LearnTabs.tsx]: Fixed in 83b90931.

File: ui-nextjs/components/learn/ModelComparisonPlayground.tsx [hunk 5]
```diff
@@ -130,12 +130,84 @@ function PipelinePanel({
...
+      setPipelineA({ parser: defaultBase, explainer: defaultBase });
+      // Find latest fine-tuned version and select both models from it
+      const ftModels = m.filter((x) => x.type === 'finetuned');
+      const latestVersion = ftModels[0]?.version ?? null;
+      if (latestVersion) {
+        const ftParser = ftModels.find((x) => x.task === 'query_parser' && x.version === latestVersion)?.model ?? defaultBase;
+        const ftExplainer = ftModels.find((x) => x.task === 'explainer' && x.version === latestVersion)?.model ?? defaultBase;
+        setPipelineB({ parser: ftParser, explainer: ftExplainer });
+      } else {
+        setPipelineB({ parser: defaultBase, explainer: defaultBase });
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply ui-nextjs/components/learn/ModelComparisonPlayground.tsx]: Fixed in f36d3a4c.

File: ui-nextjs/components/learn/ModelComparisonPlayground.tsx [hunk 6]
```diff
@@ -130,12 +130,84 @@ function PipelinePanel({
...
+      <label className="block text-[10px] text-gray-500 uppercase tracking-widest mb-1">{label}</label>
+      <select
+        value={value}
+        onChange={(e) => {
+          const model = e.target.value;
+          const info = models.find((m) => m.model === model);
+          if (info?.type === 'finetuned') {
+            onChange({
+              parser: models.find((m) => m.type === 'finetuned' && m.task === 'query_parser' && m.version === info.version)?.model ?? model,
+              explainer: models.find((m) => m.type === 'finetuned' && m.task === 'explainer' && m.version === info.version)?.model ?? model,
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply ui-nextjs/components/learn/ModelComparisonPlayground.tsx]: Fixed in f36d3a4c.

File: tests/test_analyze_endpoint.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+        mock_run = MagicMock()
+        mock_run.id = run_id
+        mock_run.task = "query_parser"
+        mock_run.metrics = {"entity_f1": 0.72}
+        mock_run.suggestions = {
+            "rules": [{"metric": "entity_f1", "severity": "blocking", "current": 0.72, "target": 0.80, "suggestion": "Add diverse entities", "category": "training_data"}],
+            "llm_analysis": "Previously analyzed",
+            "generated_at": "2026-03-27T00:00:00Z",
+        }
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply tests/test_analyze_endpoint.py]: Fixed in f36d3a4c.

File: ui-nextjs/components/learn/SuggestionsPanel.tsx [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+  return (
+    <div className="mt-4 bg-[#1a1a1a] border border-[#404040] rounded-lg p-4">
+      <h4 className="text-sm font-semibold text-white mb-3">Suggested Improvements</h4>
+
+      {blocking.length > 0 && (
+        <div className="mb-3">
+          <p className="text-xs font-medium text-red-400 uppercase tracking-wide mb-2">Blocking</p>
+          <ul className="space-y-2">
+            {blocking.map((s) => (
+              <li key={s.metric} className="text-sm bg-red-900/20 border border-red-800/30 rounded p-3">
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply ui-nextjs/components/learn/SuggestionsPanel.tsx]: Metrics are unique per task in SHIP_CRITERIA — each metric name appears exactly once per task, so key collisions can't occur.

File: ui-nextjs/components/learn/ModelComparisonPlayground.tsx [hunk 9]
```diff
@@ -130,12 +130,84 @@ function PipelinePanel({
...
+        ))}
+        {hasFt && (
+          <optgroup label="Fine-Tuned">
+            {ftVersions.map((v) => (
+              <option key={`ft-${v}`} value={models.find((m) => m.type === 'finetuned' && m.version === v)?.model ?? ''}>
+                D4BL {v ?? 'latest'}
+              </option>
+            ))}
+          </optgroup>
+        )}
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply ui-nextjs/components/learn/ModelComparisonPlayground.tsx]: The CompareResponse schema uses baseline/finetuned keys — renaming those is out of scope for this PR. The Pipeline A/B labels from the model selector already display correctly in the summary bar when explicit models are used.
